apps/prairielearn/src/lib/submissions-upload.sql

```diff
@@ -0,0 +1,91 @@
+-- BLOCK select_assessment_question
+SELECT
+  *
+FROM
+  assessment_questions
+WHERE
+  assessment_id = $assessment_id
+  AND question_id = $question_id;
+
+-- BLOCK insert_assessment_instance
+INSERT INTO
+  assessment_instances (assessment_id, user_id, number, open)
+VALUES
+  (
+    $assessment_id,
+    $user_id,
+    $instance_number,
+    false -- Assume closed by default when recreating
+  )
+RETURNING
+  id AS assessment_instance_id;
+
+-- BLOCK insert_instance_question
+INSERT INTO
+  instance_questions (
+    assessment_instance_id,
+    assessment_question_id,
+    requires_manual_grading,
+    status
+  )
+VALUES
+  (
+    $assessment_instance_id,
+    $assessment_question_id,
+    $requires_manual_grading,
+    'saved' -- Must not be the default 'unanswered' status
```

Comment:
I have a minor concern that this would not capture the result of parse, and thus would not get the difference between `saved` and `invalid`, though I don't have a potential solution here. None of the columns in the submissions CSV seems to give a clear distinction between a save-only submission and an ungradable submission. I'm ok with this assumption here for its current use, though I'm wondering if we should ensure that limitations like these and others like the lack of format_errors or raw_submitted_answer are clearly specified in the help text for this operation, to ensure we don't get complaints later from users that find this operation by accident.

Reply:
That's fair. We could of course always add extra columns to make this not as lossy, but I think just documenting the limitations for now is a good first step.